core/iwasm/compilation/aot_emit_function.c

```diff
@@ -1557,7 +1645,15 @@ aot_compile_op_call_indirect(AOTCompContext *comp_ctx, AOTFuncContext *func_ctx,
 bool
 aot_compile_op_ref_null(AOTCompContext *comp_ctx, AOTFuncContext *func_ctx)
 {
-    PUSH_I32(REF_NULL);
+#if WASM_ENABLE_GC != 0
+    if (comp_ctx->enable_gc) {
+        PUSH_REF(GC_REF_NULL);
```

Comment:
How about define PUSH_REF and POP_REF as empty macro if GC is disabled, to reduce the guarded code blocks

Reply:
Now only use the macro for necessary blocks